lib/Backend/GlobOptBailOut.cpp

```diff
@@ -15,52 +15,258 @@ GlobOpt::CaptureValue(BasicBlock *block, StackSym * stackSym, Value * value, Bai
+            }
+            else
+            {
+                StackSym* copyPropSym = this->GetCopyPropSym(block, stackSym, value);
+                if (copyPropSym)
+                {
+                    bailOutCopySymsIter.InsertNodeBefore(this->func->m_alloc, stackSym, copyPropSym);
+                }
+            }
         }
-        Sym * sym = bucket.value;
-        if (sym == nullptr || !sym->IsStackSym() || !sym->AsStackSym()->HasByteCodeRegSlot())
+        NEXT_BITSET_IN_SPARSEBV
+    }
+    else
+    {
+        // capture bailout values incrementally
+
+        CapturedValues * currCapturedValues = block->globOptData.capturedValues;
+        SListBase<ConstantStackSymValue>::Iterator iterConst(currCapturedValues ? &currCapturedValues->constantValues : nullptr);
+        SListBase<CopyPropSyms>::Iterator iterCopyPropSym(currCapturedValues ? &currCapturedValues->copyPropSyms : nullptr);
+        bool hasConstValue = currCapturedValues ? iterConst.Next() : false;
+        bool hasCopyPropSym = currCapturedValues ? iterCopyPropSym.Next() : false;
+
+        tempBv->Or(block->globOptData.changedSyms, block->globOptData.capturedArgs);
+
+        tempBv->Set(Js::Constants::InvalidSymID);
+
+        FOREACH_BITSET_IN_SPARSEBV(symId, tempBv)
         {
-            continue;
+            Sym * sym = hasConstValue ? iterConst.Data().Key() : nullptr;
+            Value * val = nullptr;
+            HashBucket<Sym *, Value *> * symIdBucket = nullptr;
+            bool symIdHandled = false;
+
+            // copy unchanged sym to new capturedValues
+            while (sym && sym->m_id < symId)
+            {
+                Assert(sym->IsStackSym());
+                if (!sym->AsStackSym()->HasArgSlotNum() || block->globOptData.capturedArgs->Test(sym->m_id))
+                {
+                    bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, sym->AsStackSym(), iterConst.Data().Value());
+                }
+
+                hasConstValue = iterConst.Next();
+                sym = hasConstValue ? iterConst.Data().Key() : nullptr;
+            }
+            if (sym && sym->m_id == symId)
+            {
+                hasConstValue = iterConst.Next();
+            }
+            if (symId != Js::Constants::InvalidSymID)
+            {
+                // recapture changed constant sym
+
+                symIdBucket = block->globOptData.symToValueMap->GetBucket(symId);
+                if (symIdBucket == nullptr)
+                {
+                    continue;
+                }
+
+                sym = symIdBucket->value;
+                Assert(sym->IsStackSym() && (sym->AsStackSym()->HasByteCodeRegSlot() || sym->AsStackSym()->HasArgSlotNum()));
+
+                val =  block->globOptData.capturedArgs->Test(symId) ? FindValue(sym) : symIdBucket->element;
+                ValueInfo* valueInfo = val->GetValueInfo();
+
+                if (valueInfo->GetSymStore() != nullptr)
```

Comment:
This check is inherited from the previous version.

---
In reply to: [68345902](https://github.com/Microsoft/ChakraCore/pull/1152#discussion_r68345902) [](ancestors = 68345902)